sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -453,3 +455,119 @@ case class Murmur3Hash(children: Seq[Expression], seed: Int) extends Expression
     }
   }
 }
+
+/**
+ * A function that encrypts input using AES. Key lengths of 128, 192 or 256 bits can be used.
+ * 192 and 256 bits keys can be used if Java Cryptography Extension (JCE) Unlimited Strength
+ * Jurisdiction Policy Files are installed. If either argument is NULL or the key length is
+ * not one of the permitted values, the return value is NULL.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(input, key) - Encrypts input using AES.",
+  extended = "> SELECT Base64(_FUNC_('ABC', '1234567890123456'));\n 'y6Ss+zCYObpCbgfWfyNWTw=='")
+case class AesEncrypt(left: Expression, right: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes {
+
+  override def dataType: DataType = BinaryType
+  override def nullable: Boolean = true
+
+  override def inputTypes: Seq[DataType] = Seq(BinaryType, BinaryType)
+
+  protected override def nullSafeEval(input1: Any, input2: Any): Any = {
+    val cipher = Cipher.getInstance("AES")
+    val secretKey: SecretKeySpec = input2.asInstanceOf[Array[Byte]].length match {
+      case 16 | 24 | 32 =>
+        new SecretKeySpec(input2.asInstanceOf[Array[Byte]], 0,
+          input2.asInstanceOf[Array[Byte]].length, "AES")
+      case _ => null
+    }
+
+    try {
+      cipher.init(Cipher.ENCRYPT_MODE, secretKey)
+      cipher.doFinal(input1.asInstanceOf[Array[Byte]], 0, input1.asInstanceOf[Array[Byte]].length)
+    } catch {
+      case e: GeneralSecurityException => null
```

Comment:
@cloud-fan What I have implemented was trying to follow the way `Sha2` did. 
I think if secret key is invalid. It will return null just like the way `Sha2` did. There is no other errors should be thrown during execution.